### PR TITLE
Add GH Action for commit message validation.

### PR DESCRIPTION
**Pull request by OhmSpectator**
This PR introduces a new GitHub Actions workflow to automatically lint commit messages, ensuring they adhere to the recommended format. The workflow is triggered on pushes and pull requests to specific branches, including master and version branches. It includes a Python script that checks for the correct commit message format, ensuring that each message has a subject, an empty line separating the subject from the body, and a non-empty body. Additionally, the PR adds a requirements.txt file for managing dependencies and updates the .gitignore file to exclude the virtual environment directory. This enhancement helps maintain consistent and readable commit messages across the repository.

### CODE REVIEW

File: tools/check-commit-messages/check_commit_messages.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+"""
+Copyright (c) 2024, Zededa Inc.
+SPDX-License-Identifier: Apache-2.0
+
+This script checks the commit messages in the current branch to ensure they follow the
+recommended format. The recommended format is as follows:
+* Commit message should have a subject and body
+* An empty line should separate the subject and body
+* The body should not be empty (it should contain more than just "Signed-off-by")
```
Comment (milan-zededa) [inline tools/check-commit-messages/check_commit_messages.py:9]: Should we also impose line length limit?

**Review by milan-zededa — APPROVED** [review]
Thanks for both new linters, you had a very productive night it seems :)

File: tools/check-commit-messages/check_commit_messages.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+
+def main():
+    """
+    Main function to check the commit messages in the current branch.
+    """
+    repo = git.Repo(search_parent_directories=True)
+
+    # Fetch the latest from origin/master to ensure we are comparing correctly
+    repo.remotes.origin.fetch('master')
```
Comment (milan-zededa) [inline tools/check-commit-messages/check_commit_messages.py:54]: Same suggestion as posted here: https://github.com/lf-edge/eve/pull/4166/files#r1732333183

File: tools/check-commit-messages/check_commit_messages.py (hunk 1 above)
Comment (OhmSpectator) [reply tools/check-commit-messages/check_commit_messages.py:9]: I thought about it, but we have too many people for whom even these rules can be too strict =D So, let's introduce the restrictions step by step.

File: tools/check-commit-messages/check_commit_messages.py (hunk 2 above)
Comment (OhmSpectator) [reply tools/check-commit-messages/check_commit_messages.py:54]: Yep, will have to fix it.

File: tools/check-commit-messages/check_commit_messages.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    commits = list(repo.iter_commits('origin/master..HEAD'))
```
[the remark this answers is not in the document]
Comment (OhmSpectator) [reply tools/check-commit-messages/check_commit_messages.py:56]: Most probably not. I will switch to the GH variable.

**Comment by rene** [comment]
Thanks for this PR @OhmSpectator , I think we were missing for a long time this feature, definitely it will help (enforce :smile: )  us to standardize and improve our commit messages.... I have just one comment, it could also check for a capitalized first word of the body message, as asked in our guidelines: https://github.com/lf-edge/eve/blob/master/CONTRIBUTING.md#commit-messages , but it can be done in a further iteration as well...

**Comment by OhmSpectator** [comment]
> I have just one comment, it could also check for a capitalized first word of the body message, as asked in our guidelines: https://github.com/lf-edge/eve/blob/master/CONTRIBUTING.md#commit-messages ,

Lol, I have never seen these guidelines =D
Will add a check now.

**Comment by OhmSpectator** [comment]
@rene, I've added the capital letter check.

**Comment by rene** [comment]
Thanks again @OhmSpectator , I'm approving this PR but I would like to drop one more idea here: it would be really nice to integrate this check as a git hook, you can provide a custom script to check commit message in `.git/hooks/commit-msg`, maybe for a future PR....

**Review by rene — APPROVED** [review]

**Comment by OhmSpectator** [comment]
Yeah, let's add a hool the next step. Let the team first get used to the rules by seeing failures in the GH Actions =)

**Pull request merged**